### PR TITLE
Sensible Defaults for Augmented 6th

### LINKED ISSUE #529: Change Spelling of Aug6 inversions in some cases
Thinking through Music21's handling of Aug6th RomanNumerals, here’s my thought of refining it (in c minor):

Any of these can have a + added after the country name.

The “proper” spelling OR the country name alone OR the country name with a 6 afterwards becomes the standard inversion of the chord:

Ger, Ger6, Ger65 — all = standard German 6th.  Ab in bass — changed to Ger65 internally
Ger43 = C in bass
Ger42 = Eb in bass
Ger7 = F# in bass  (n.b., this is not Ger7, not Ger43)

Fr, Fr6, Fr43 — all = standard French 6th.  Ab in bass — changed to Fr43 internally
Fr42 = C in bass
Fr7 = D in bass
Fr65 = F# in bass

Sw/Swiss/English/whatever you call sharp-2 — behaves exactly as French.

Italian is tricker, because there are two things that “It” alone might mean, so we will require:

It, It6 = standard Italian 6th.  Ab in bass.  Changed to It6 internally
It64 = C in bass
It53 = F# in bass — this is the ONLY case where we require 53 to be specified.

For Ger7 possibly allow Gero3 or Gerdim3 

Current music21 output:

```
In [14]: for c in ('Ger', 'Ger6', 'Ger65', 'Ger43', 'Ger42', 'Ger7', 'Fr', 'Fr6', 'Fr43', 'Fr42', 'Fr7', 'Fr65', 
    ...: 'Sw', 'Sw6', 'Sw43', 'Sw42', 'Sw7', 'Sw65', 'It', 'It6', 'It64', 'It53', 'It7'): 
    ...:     try: 
    ...:         r = roman.RomanNumeral(c, 'C') 
    ...:         print(r, '\t',  
    ...:             ' '.join(p.nameWithOctave for p in r.pitches)) 
    ...:     except: 
    ...:         print(f'No dice for {r}') 
```

with stars next to ones that need updating:

```
* <music21.roman.RomanNumeral Ger in C major> 	 F#4 A-4 C5
* <music21.roman.RomanNumeral Ger6 in C major> 	 A-4 C5 F#5
  <music21.roman.RomanNumeral Ger65 in C major>  A-4 C5 E-5 F#5
  <music21.roman.RomanNumeral Ger43 in C major>  C4 E-4 F#4 A-4
  <music21.roman.RomanNumeral Ger42 in C major>  E-4 F#4 A-4 C5
  <music21.roman.RomanNumeral Ger7 in C major> 	 F#4 A-4 C5 E-5
* <music21.roman.RomanNumeral Fr in C major> 	 D4 F#4 A-4
* <music21.roman.RomanNumeral Fr6 in C major> 	 F#4 A-4 D5
  <music21.roman.RomanNumeral Fr43 in C major> 	 A-4 C5 D5 F#5
  <music21.roman.RomanNumeral Fr42 in C major> 	 C4 D4 F#4 A-4
  <music21.roman.RomanNumeral Fr7 in C major> 	 D4 F#4 A-4 C5
  <music21.roman.RomanNumeral Fr65 in C major> 	 F#4 A-4 C5 D5
* <music21.roman.RomanNumeral Sw in C major> 	 D#4 F#4 A-4
* <music21.roman.RomanNumeral Sw6 in C major> 	 F#4 A-4 D#5
  <music21.roman.RomanNumeral Sw43 in C major> 	 A-4 C5 D#5 F#5
  <music21.roman.RomanNumeral Sw42 in C major> 	 C4 D#4 F#4 A-4
  <music21.roman.RomanNumeral Sw7 in C major> 	 D#4 F#4 A-4 C5
  <music21.roman.RomanNumeral Sw65 in C major> 	 F#4 A-4 C5 D#5
* <music21.roman.RomanNumeral It in C major> 	 F#4 A-4 C5
  <music21.roman.RomanNumeral It6 in C major> 	 A-4 C5 F#5
  <music21.roman.RomanNumeral It64 in C major> 	 C4 F#4 A-4
  <music21.roman.RomanNumeral It53 in C major> 	 F#4 A-4 C5
* <music21.roman.RomanNumeral It7 in C major> 	 F#4 A-4 C5 E-5
```

### PR DESCRIPTION
**Pull request by mscuthbert**
Fixes #529

Diagnoses that there are larger problems with postFigureFromChordAndKey  not taking into account major and minor differences.  It needs a major refactor  BUT this is a step in the right direction and for helping give many better test cases

Also fixes a small bug in Mordents and intervals in Expressions and adds some typing in makeNotation -- it's not related but I accidentally had it sitting around the same directory.

### CODE REVIEW

File: music21/pitch.py [hunk 1]
```diff
@@ -2912,6 +2912,10 @@ def pitchClass(self) -> int:
         >>> a.pitchClass = 1
         >>> a
         <music21.pitch.Pitch C#3(+20c)>
+
+        OMIT FROM DOCS
+
+        this is a performance critical operation.
         '''
         pc = round(self.ps) % 12
         if pc == 12:
```
Comment (jacobtylerwalls) [inline music21/pitch.py:2921]: speaking of performance, what is `if pc == 12` doing here? was modulo different in Python 2? 🤔

File: music21/chord/__init__.py [hunk 2]
```diff
@@ -2654,12 +2619,11 @@ def isItalianAugmentedSixth(self, restrictDoublings=False) -> bool:
         >>> c4.isItalianAugmentedSixth()
         False
 
-        If inversions don't matter to you, put the chord in another inversion:
+        If inversions don't matter to you, add `permitAnyInversion=Tru`:
```
Comment (jacobtylerwalls) [inline music21/chord/__init__.py:2622]: True

File: music21/pitch.py [hunk 3]
```diff
@@ -2912,6 +2912,10 @@ def pitchClass(self) -> int:
         >>> a.pitchClass = 1
         >>> a
         <music21.pitch.Pitch C#3(+20c)>
+
+        OMIT FROM DOCS
```
Comment (jacobtylerwalls) [inline music21/pitch.py:2916]: think this needs underscores

File: music21/pitch.py (hunk 1 above)
Comment (mscuthbert) [reply music21/pitch.py:2921]: OMG -- I totally forgot about this bug.  Yes!  It was possible before Python 2.5 to have negative ps's become 12 so we had to do more complex things to fix it.  Totally impossible for years now!  But we need to cast it to an int -- so there IS a bug here.

File: music21/pitch.py (hunk 3 above)
Comment (mscuthbert) [reply music21/pitch.py:2916]: THANKS!

File: music21/chord/__init__.py (hunk 2 above)
Comment (mscuthbert) [reply music21/chord/__init__.py:2622]: Thanks!

**Review by mscuthbert — COMMENTED** [review]
Thanks Jacob!

File: music21/pitch.py (hunk 1 above)
Comment (mscuthbert) [reply music21/pitch.py:2921]: No wait, it doesn't need to cast to an int.  That's only because I was playing around in Python 2 just now.  Good riddance!

**Pull request merged**